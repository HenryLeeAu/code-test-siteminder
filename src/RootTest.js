// reuse redux provider in real component and testing code
import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import reducers from 'reducers';
import ReduxThunk from 'redux-thunk';

export default ({ children, initialState = {} }) => {
  const store = createStore(
    reducers,
    initialState,
    applyMiddleware(ReduxThunk)
  );

  return <Provider store={store}>{children}</Provider>;
};
