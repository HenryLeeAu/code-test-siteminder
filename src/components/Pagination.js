import React, { Component } from "react";
import { connect } from "react-redux";
import { increaseCurrentPage,decreaseCurrentPage } from "actions";
class Pagination extends Component {
  componentDidMount() {
    //console.log(this.props.searchStatus)
  }
  handleClickDecreasment=(e)=>{
    this.props.decreaseCurrentPage()
   
  }
  handleClickIncreasment=(e)=>{
    this.props.increaseCurrentPage()
    
  }
  
  renderPagination(){
    const { currentPage, totalPages,totalNum } = this.props.searchStatus;
    if(totalPages!==null && totalPages>1 ){
      return (
        <div className="pagination">
        <div className="left">
          <button id="dec" onClick= { this.handleClickDecreasment}>prev</button>
        </div>
        <div className="middle">
          <div>  {currentPage} / {totalPages} pages</div>
          {totalNum} total
        </div>
        <div className="right">
          <button onClick= { this.handleClickIncreasment}>next</button>
        </div>
      </div>
      )
    }else{
      return null
    }
    
  }
  render() {
    
    return  this.renderPagination()
  }
}

function mapStateToProps(state) {
  return {
    searchStatus: state.searchStatus
  };
}

export default connect(
  mapStateToProps,
  {increaseCurrentPage,decreaseCurrentPage}
)(Pagination);
