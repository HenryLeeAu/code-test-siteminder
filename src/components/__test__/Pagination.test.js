import React from "react";

import Pagination from "components/Pagination";
import { mount, unmount } from "enzyme";
import RootTest from "RootTest";
describe("Pagination", () => {
  it("more than 1 pages", () => {
    const initialState = {
      searchStatus: {
        totalPages:4,
        currentPage:1,
        totalNum:40
      }
    }
    const wrappered = mount(
      <RootTest initialState={initialState}>
        <Pagination />
      </RootTest>
    );
    expect(wrappered).toMatchSnapshot();
    wrappered.unmount();
  });
  it("less than 1 page", () => {
    const initialState = {
      searchStatus: {
        totalPages:null,
        currentPage:null,
        totalNum:null
      }
    }
    const wrappered = mount(
      <RootTest >
        <Pagination initialState={initialState} />
      </RootTest>
    );
    expect(wrappered).toMatchSnapshot();
    wrappered.unmount();
  });
});