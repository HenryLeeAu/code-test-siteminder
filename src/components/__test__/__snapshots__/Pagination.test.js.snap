// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination less than 1 page 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(Pagination)
      initialState={
        Object {
          "searchStatus": Object {
            "currentPage": null,
            "totalNum": null,
            "totalPages": null,
          },
        }
      }
    >
      <Pagination
        decreaseCurrentPage={[Function]}
        increaseCurrentPage={[Function]}
        initialState={
          Object {
            "searchStatus": Object {
              "currentPage": null,
              "totalNum": null,
              "totalPages": null,
            },
          }
        }
        searchStatus={
          Object {
            "currentId": null,
            "currentMovieDetail": null,
            "currentMovieList": Array [],
            "currentPage": null,
            "keyword": "",
            "totalNum": null,
            "totalPages": null,
            "visitedMovieDetailList": Array [],
          }
        }
      />
    </Connect(Pagination)>
  </Provider>
</_default>
`;

exports[`Pagination more than 1 pages 1`] = `
<_default
  initialState={
    Object {
      "searchStatus": Object {
        "currentPage": 1,
        "totalNum": 40,
        "totalPages": 4,
      },
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(Pagination)>
      <Pagination
        decreaseCurrentPage={[Function]}
        increaseCurrentPage={[Function]}
        searchStatus={
          Object {
            "currentPage": 1,
            "totalNum": 40,
            "totalPages": 4,
          }
        }
      >
        <div
          className="pagination"
        >
          <div
            className="left"
          >
            <button
              id="dec"
              onClick={[Function]}
            >
              prev
            </button>
          </div>
          <div
            className="middle"
          >
            <div>
                
              1
               / 
              4
               pages
            </div>
            40
             total
          </div>
          <div
            className="right"
          >
            <button
              onClick={[Function]}
            >
              next
            </button>
          </div>
        </div>
      </Pagination>
    </Connect(Pagination)>
  </Provider>
</_default>
`;
